core/src/main/scala/kafka/server/OffsetCheckpoint.scala

```diff
@@ -41,7 +41,11 @@ class OffsetCheckpoint(val file: File) extends Logging {
   private val path = file.toPath.toAbsolutePath
   private val tempPath = Paths.get(path.toString + ".tmp")
   private val lock = new Object()
-  file.createNewFile() // in case the file doesn't exist
+  try {
+    Files.createFile(file.toPath) // create the file if it doesn't exist
+  } catch {
+    case e :FileAlreadyExistsException => None
```

Comment:
`case _: FileAlreadyExistsException => ()`. There are a few others like this one.